### PR TITLE
adds explicit initialization and peek (namespace) handlers to lib/tapp

### PR DESCRIPTION
**Pull request by joemfb**
And a couple default namespace paths for inspecting the state of a tapp.

### CODE REVIEW

File: lib/tapp.hoon [hunk 1]
```diff
@@ -28,6 +30,8 @@
       active=(unit eval-form:eval:tapp-async)
       app-state=state-type
   ==
++$  tapp-peek
+  [%noun ?(? (set contract))]
```
Comment (jtobin) [inline lib/tapp.hoon]: My mental parser paused for a second here to verify that `?(?` wasn't some strange irregular syntax it hadn't seen before.

**Comment by philipcmonk** [comment]
I don't have ny github password on my phone so I can't mark it approved
right now, but this looks good to me.

File: lib/tapp.hoon [hunk 2]
```diff
@@ -44,7 +48,11 @@
   %-  create-tapp-poke-peer
   |_  [=bowl:gall state=state-type]
   ++  handle-poke  ~(handle-poke handler bowl state)
-  ++  handle-peer  |=(* (async-fail:async-lib %no-peer-handler >path< ~))
+  ++  handle-peer
+    |=  *
+    ~|  %default-tapp-no-sole
+    ?<  ?=([%sole *] +<)
```
Comment (jtobin) [inline lib/tapp.hoon]: As an anti-larkite, I would give the sample a face and kill the lark here:

```
++  handle-peer
   |=  =noun
    ~|  %default-tapp-no-sole
    ?<  ?=([%sole *] noun)
   (async-fail:async-lib %no-peer-handler >path< ~)
```

**Comment by jtobin** [comment]
🚢

**Pull request merged**